storage/SystemInfo/CmdParted.cc

```diff
@@ -125,11 +135,13 @@ namespace storage
 		string tmp = extractNthWord(0, *it);
 		if (!tmp.empty() && isdigit(tmp[0]))
 		{
-		    assert(n == 1 || n == 2);
 		    if (n == 1)
 			scanCylEntryLine(*it);
 		    else if (n == 2)
 			scanSecEntryLine(*it);
+		    else
+			ST_THROW( ParseException( string( "Unexpected partition table #" )
+						  + std::to_string(n), "", "" ) );
```

Comment:
Well, yes, but the problem here is that the offending line (the line that caused the parse error) is not exact to identify here: It would have been the starting line of that excess partition table, not just any random entry in it. But that line is already gone at this point.